Sources/ZippyJSON/ZippyJSONDecoder.swift

```diff
@@ -1057,3 +1078,63 @@ extension __JSONDecoder : SingleValueDecodingContainer {
 
     // End
 }
+
+@propertyWrapper
+private final class Synchronized<Value>: @unchecked Sendable {
+    private let lock: LockProtocol
```

Comment:
so it appears that this design creates like 4 of these per decoder? i believe that should be fast, but i may verify